lib/coinbase/address/wallet_address.rb

```diff
@@ -184,12 +184,34 @@ def ensure_sufficient_balance!(amount, asset_id)
       raise InsufficientFundsError.new(amount, current_balance)
     end
 
-    def complete_staking_operation(amount, asset_id, action, mode: :default, options: {})
+    def complete_staking_operation(amount, asset_id, action, mode: :default, options: {}, interval_seconds: 5,
+                                   timeout_seconds: 600)
       op = StakingOperation.create(amount, network, asset_id, id, wallet_id, action, mode, options)
-      op.transactions.each do |transaction|
-        transaction.sign(@key)
+      start_time = Time.now
+
+      while Time.now - start_time < timeout_seconds
```

Comment:
Logic like this should not be in the `WalletAddress` model.

Let's move this to be a `StakingOperation` concern

Reply:
Done 👍